src/VisualStudio/Core/Def/Implementation/LanguageService/AbstractPackage`2.cs

```diff
@@ -119,15 +133,15 @@ internal IComponentModel ComponentModel
         protected abstract IEnumerable<IVsEditorFactory> CreateEditorFactories();
         protected abstract TLanguageService CreateLanguageService();
 
-        protected void RegisterService<T>(Func<T> serviceCreator)
+        protected void RegisterService<T>(Func<CancellationToken, Task<T>> serviceCreator)
```

Comment:
Does this break F#/TypeScript?

Reply:
It might. We would need time to update those for whenever we plan to merge this.